src/scancode/plugin_ignore.py

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (c) 2017 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from functools import partial
+
+from commoncode.ignore import is_ignored
+from plugincode.pre_scan import PreScanPlugin
+from plugincode.pre_scan import pre_scan_impl
+
+
+@pre_scan_impl
+class ProcessIgnore(PreScanPlugin):
+    """
+    Ignore files matching <pattern>.
+    """
+    option_attrs = dict(multiple=True, metavar='<pattern>')
+
+    def __init__(self, user_input):
+        super(ProcessIgnore, self).__init__(user_input)
+
+    def process_resources(self, resources):
+        user_ignore = {patt: 'User ignore: Supplied by --ignore' for patt in self.user_input}
+        ignored = partial(is_ignored, ignores=user_ignore, unignores={})
```

Comment:
No, these are applied after the resources are processed for ignore.ignores_VCS.
Here https://github.com/nexB/scancode-toolkit/blob/8a7b660821dd6cfe7c1c077ccc052225044bb9c2/src/scancode/cli.py#L776
The `resources` are passed to `process_resources` function of a plugin after all the default processing is done